filemanager/s3manager.go

```diff
@@ -0,0 +1,313 @@
+package filemanager
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/url"
+	"os"
+	"path"
+	"strings"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/s3"
+	awsS3Manager "github.com/aws/aws-sdk-go/service/s3/s3manager"
+	"github.com/mitchellh/mapstructure"
+	"github.com/samber/lo"
+
+	"github.com/rudderlabs/rudder-go-kit/awsutil"
+	appConfig "github.com/rudderlabs/rudder-go-kit/config"
+	"github.com/rudderlabs/rudder-go-kit/logger"
+)
+
+type S3Config struct {
+	Bucket           string  `mapstructure:"bucketName"`
+	Prefix           string  `mapstructure:"Prefix"`
+	Region           *string `mapstructure:"region"`
+	Endpoint         *string `mapstructure:"endpoint"`
+	S3ForcePathStyle *bool   `mapstructure:"s3ForcePathStyle"`
+	DisableSSL       *bool   `mapstructure:"disableSSL"`
+	EnableSSE        bool    `mapstructure:"enableSSE"`
+	RegionHint       string  `mapstructure:"regionHint"`
+	UseGlue          bool    `mapstructure:"useGlue"`
+}
+
+// NewS3Manager creates a new file manager for S3
+func NewS3Manager(config map[string]interface{}, log logger.Logger, defaultTimeout func() time.Duration) (*s3Manager, error) {
+	var s3Config S3Config
+	if err := mapstructure.Decode(config, &s3Config); err != nil {
+		return nil, err
+	}
+	regionHint := appConfig.GetString("AWS_S3_REGION_HINT", "us-east-1")
```

Comment:
why are we not getting this from config too?

Reply:
Didn't change the previous behaviour, I suspect that region hint is not part of the destination's specific configuration? Not sure though